packages/front-end/components/SchemaBrowser/EditSqlModal.tsx

```diff
@@ -96,16 +96,22 @@ export default function EditSqlModal({
       header="Edit SQL"
       submit={form.handleSubmit(async (value) => {
         if (testQueryBeforeSaving) {
-          const res = await runTestQuery(value.sql);
+          let res: TestQueryResults;
+          try {
+            res = await runTestQuery(value.sql);
+          } catch (e) {
+            setTestQueryResults({ sql: value.sql, error: e.message });
+            return Promise.reject(e.message);
+          }
           if (res.error) {
-            throw new Error(res.error);
+            setTestQueryResults(res);
+            return Promise.reject(res.error);
```

Comment:
This did not show up because I wasn't throwing a real error but just the message.  I've change this to throw a real error, but have the message be blank so it also doesn't show, but more for the right reasons.